### DISCUSSION TITLE
Add Naoris protocol security to akash

### DISCUSSION THREAD

**Discussion by 31trainman** [akash-network/community#350] [Integrations]
Naoris protocol is a decentralized mesh security platform. I feel it would add value to akash network by offering better security than what we have now. The Naoris protocol adds value to the akash network by providing a security system to protect network nodes etc. By bringing integration with Naoris protocol I feel it will offer akash network users better cybersecurity for their hosted workloads etc.

Here is the website of Naoris protocol: https://naorisprotocol.com/

**Comment by hewi333** [comment]
Hi there 31trainman! Thanks for your post. Can you go into a little more detail on Naoris, how it works, what level of integration you are proposing, etc. The whitepaper requires an email to view.

More specifically,
How does Naoris offer better security than we have now?
Also, has this project launched or can you give examples of how it is providing better security today?

**Reply by 31trainman** [reply]
Naoris protocol offers better security by turning each node we run into a security validator by verifying that each node is providing data that is free of viruses and bugs. This is done using a mesh structure. Eliminating any onenode being a point of failure for network security.

From what I can tell the Naoris protocol is still under development and not fully operational

**Reply by hewi333** [reply]
In proof-of-stake networks such as Akash, the security is derived from validators reaching consensus by putting value at stake. If any validator acts maliciously, the majority of honest validators work to prevent and correct any nefarious behavior. In our case, we have 100 validators in the active set. If one is acting maliciously, there are 99 others so there is no single point of failure.

If Naoris protocol is still under development and not fully operational, how can we add it? Are you looking for integration funding or just bringing to light something that may be helpful down the road once it's developed?

**Reply by 31trainman** [reply]
Future integration. I wasn't aware at the time of my posting that naoris protocol wasn't fully ready yet.
